File: backend/components/metiers/threed/tests_centralises/legacy/samples/rbac/test_rbac_sample.py
"""
Test ultra avancé pour rbac_sample.py
"""

from threed.legacy.samples.rbac.rbac_sample import rbac_sample


def test_rbac_admin():
    assert "tous les droits" in rbac_sample("alice", "admin")


def test_rbac_user():
    assert "accès restreint" in rbac_sample("bob", "user")


def test_rbac_empty_user():
    assert "tous les droits" in rbac_sample("", "admin")
    assert "accès restreint" in rbac_sample("", "user")


def test_rbac_none_user():
    assert "tous les droits" in rbac_sample(None, "admin")
    assert "accès restreint" in rbac_sample(None, "user")


def test_rbac_unknown_role():
    assert "accès restreint" in rbac_sample("eve", "guest")


# ...autres tests avancés, edge cases, sécurité...

File: backend/components/metiers/threed/tests_centralises/legacy/samples/rbac/test_rbac_sample.js
// Test ultra avancé pour rbac_sample.js
const rbacSample = require('../../../legacy/samples/rbac/rbac_sample');

describe('RBAC Sample', () => {
  it('doit donner tous les droits à admin', () => {
    expect(rbacSample('alice', 'admin')).toMatch('tous les droits');
  });
  it('doit restreindre un user non admin', () => {
    expect(rbacSample('bob', 'user')).toMatch('accès restreint');
  });
  it('doit gérer un user vide', () => {
    expect(rbacSample('', 'admin')).toMatch('tous les droits');
    expect(rbacSample('', 'user')).toMatch('accès restreint');
  });
  it('doit gérer un user null', () => {
    expect(rbacSample(null, 'admin')).toMatch('tous les droits');
    expect(rbacSample(null, 'user')).toMatch('accès restreint');
  });
  it('doit gérer un rôle inconnu', () => {
    expect(rbacSample('eve', 'guest')).toMatch('accès restreint');
  });
  // ...autres tests avancés, edge cases, sécurité...
});
